packages/logging/src/logging-and-telemetry.ts

```diff
@@ -160,26 +127,29 @@ export class LoggingAndTelemetry implements MongoshLoggingAndTelemetry {
   }
 
   public flush(): void {
-    // Run any telemetry events even if device ID hasn't been resolved yet
-    this.runAndClearPendingTelemetryEvents();
-
     // Run any other pending events with the set or dummy log for telemetry purposes.
     this.runAndClearPendingBusEvents();
 
-    this.resolveDeviceId('unknown');
+    // Abort setup, which will cause the device ID to be set to 'unknown'
+    // and run any remaining telemetry events
+    this.telemetrySetupAbort.abort();
   }
 
   private async setupTelemetry(): Promise<void> {
     if (!this.deviceId) {
-      this.deviceId = await Promise.race([
-        getDeviceId({
-          onError: (error) =>
-            this.bus.emit('mongosh:error', error, 'telemetry'),
-        }),
-        new Promise<string>((resolve) => {
-          this.resolveDeviceId = resolve;
-        }),
-      ]);
+      // eslint-disable-next-line @typescript-eslint/no-var-requires
+      const getMachineId = require('native-machine-id').getMachineId;
```

Comment:
This was previously in a try/catch (to avoid crashing mongosh in situations in which we cannot load this), should it continue to be?